src/pages/test-report/components/legal-requirements-modal/legal-requirements-modal.ts

```diff
@@ -0,0 +1,32 @@
+import { Component } from '@angular/core';
+import { IonicPage, NavParams } from 'ionic-angular';
+import { CatBLegalRequirements } from '../../../../modules/tests/test-data/test-data.models';
+import { legalRequirementsLabels } from '../../../../shared/constants/legal-requirements/catb-legal-requirements';
+
+@IonicPage()
+@Component({
+  selector: 'legal-requirements-modal',
+  templateUrl: 'legal-requirements-modal.html',
+})
+export class LegalRequirementsModal {
+
+  onCancel: Function;
+  onTerminate: Function;
+  // TODO - Change the type to make this generic for other test categories
+  legalRequirements: CatBLegalRequirements;
+
+  constructor(
+    private navParams: NavParams,
+  ) {
+    this.onCancel = this.navParams.get('onCancel');
```

Comment:
Instead of passing callbacks around could we just use the ViewController to dismiss the modal from within, like so -
```
  constructor(
    private viewCtrl: ViewController,
    private navParams: NavParams,
  ) {
    this.legalRequirements = this.navParams.get('legalRequirements');
  }

  onCancel() {
    this.viewCtrl.dismiss();
  }

  onTerminate() {
    this.viewCtrl.dismiss();
    // TODO - MES-59 to handle terminate test page
    // .then(() => this.navCtrl.push('TerminateTestPage'));
  }
```

Reply:
Hmm, I think the problem with that approach is that the nav controller is needed for termination (and continue actions on the End Test modal) and it adds the pages to the modal view, not the main app view

Reply:
Unless we did something like
```this.viewCtrl.dismiss(action);```

and on test report page:
```this.modal = this.modalController.create('LegalRequirementsModal',
  this.modal.onDidDismiss(data => {
     if (action === 'continue') {
       this.navCtrl.push('DebriefPage');
     }
});
modal.present();
```